src/pages/Product/Product.layout.vue

```diff
@@ -0,0 +1,34 @@
+<template>
+  <div class="Product__container">
+    <h1 class="Product__name">{{ product.name }}</h1>
+    <q-img class="Product__image" :src="product.image_url" />
+    <div class="Product__details">
+      <div>
+        <span class="Product__price">S/. {{ product.base_price_sol }}</span>
+        <p class="Product__description">{{ product.description }}</p>
+      </div>
+      <q-btn
+        class="Product__cart_button"
+        color="accent"
+        icon="add_shopping_cart"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { product } from 'src/utils';
+import { defineComponent, PropType } from 'vue';
+
+export default defineComponent({
+  name: 'Product',
+  props: {
+    product: {
+      type: Object as PropType<product>,
+      default: () => {
```

Comment:
Why the default is a function?

Reply:
It's a factory function, the idea is to return the object that we want as "default".

Here's a comment that explains why on VueJS this is required

https://forum.vuejs.org/t/why-component-prop-default-must-be-returned-from-a-factory-function/100950/2

On resume, it's to avoid the mutation of the object/array, seems like VueJs is calling the function that generates the new Object each time that the object is called

Reply:
HMm OK, its confusing but I get the idea hehe Also factory functions only work like that in vue, I think in plain JS you only assign an empty `{}` as default 🤷🏻‍♂️